indexer/lib/chainalysis.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Check a single address with the Chainalysis API
+ *
+ * @param {string} apiKey - Chainalysis API key
+ * @param {string} address - Ethereum address to check
+ * @param {Object} options - Additional options
+ * @param {Function} [options.fetch] - Custom fetch function for testing
+ * @returns {Promise<boolean | null>}
+ */
+export async function isAddressSanctioned(
+  apiKey,
+  address,
+  { fetch = globalThis.fetch } = {},
+) {
+  try {
+    const url = `https://public.chainalysis.com/api/v1/address/${address}`
+
+    const response = await fetch(url, {
+      method: 'GET',
+      headers: {
+        'X-API-KEY': apiKey,
+        Accept: 'application/json',
+      },
+    })
+
+    if (!response.ok) {
+      console.error(
+        `Chainalysis API error: ${response.status} ${response.statusText}`,
+      )
+
+      return null
```

Comment:
In this case, we can store a sanctioned payer as not sanctioned, which is risky. I think we have to retry, until the chainalysis call worked. Ideally, we create a queue for this, so retries are reliable. I'm not sure what the best approach for M2 is though 🤔 We also don't want to not index proof sets, when chainalysis is unavailable, and the payer is not sanctioned.

@bajtos wdyt?

Reply:
There seems to be a [Cloudflare Queue](https://developers.cloudflare.com/queues/configuration/batching-retries/) product that could be integrated together with the Cloudflare Worker to handle retries.

Reply:
I have added the retry process in this commit https://github.com/filcdn/worker/pull/171/commits/13e6e870e23337a9d54601607d60de9d76bd28f3. Maybe it would be a good idea to add some limits on how many times the message should be retried before we drop it or send it to some other queue?

Reply:
+1 to add a retry queue 👍🏻